src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Operations/CosmosFhirOperationDataStore.cs

```diff
@@ -0,0 +1,227 @@
+            if (eTag != null)
+            {
+                requestOptions.AccessCondition = new AccessCondition()
+                {
+                    Type = AccessConditionType.IfMatch,
+                    Condition = eTag.VersionId,
+                };
+            }
+
+            try
+            {
+                ResourceResponse<Document> replaceResult = await DocumentClient.ReplaceDocumentAsync(
+                    UriFactory.CreateDocumentUri(DatabaseId, CollectionId, jobRecord.Id),
+                    cosmosExportJob,
+                    requestOptions,
+                    cancellationToken: cancellationToken);
+
+                var latestETag = replaceResult.Resource.ETag;
+                return new ExportJobOutcome(jobRecord, WeakETag.FromVersionId(latestETag));
+            }
+            catch (DocumentClientException dce)
+            {
+                if (dce.StatusCode == HttpStatusCode.TooManyRequests)
+                {
+                    throw new RequestRateExceededException(dce.RetryAfter);
+                }
+                else if (dce.StatusCode == HttpStatusCode.PreconditionFailed)
+                {
+                    throw new JobConflictException();
+                }
+                else if (dce.StatusCode == HttpStatusCode.NotFound)
+                {
+                    throw new JobNotFoundException(string.Format(Core.Resources.JobNotFound, jobRecord.Id));
+                }
+
+                _logger.LogError(dce, "Unhandled Document Client Exception");
+                throw;
+            }
+        }
+
+        public async Task<IReadOnlyCollection<ExportJobOutcome>> AcquireExportJobsAsync(
+            ushort maximumNumberOfConcurrentJobsAllowed,
+            TimeSpan jobHeartbeatTimeoutThreshold,
+            CancellationToken cancellationToken)
+        {
+            try
+            {
+                StoredProcedureResponse<IReadOnlyCollection<CosmosExportJobRecordWrapper>> response = await _retryExceptionPolicyFactory.CreateRetryPolicy().ExecuteAsync(
+                    async ct => await _acquireExportJobs.ExecuteAsync(
+                        DocumentClient,
+                        CollectionUri,
+                        maximumNumberOfConcurrentJobsAllowed,
+                        (ushort)jobHeartbeatTimeoutThreshold.TotalSeconds,
+                        ct),
+                    cancellationToken);
+
+                return response.Response.Select(wrapper => new ExportJobOutcome(wrapper.JobRecord, WeakETag.FromVersionId(wrapper.ETag))).ToList();
+            }
+            catch (DocumentClientException dce)
+            {
+                string subStatusInString = dce.ResponseHeaders.Get(CosmosDbHeaders.SubStatus);
+
+                if (!string.IsNullOrEmpty(subStatusInString) &&
+                    int.TryParse(subStatusInString, NumberStyles.Integer, CultureInfo.InvariantCulture, out int subStatus))
```

Comment:
I just opened PR #449 to apply this pattern across out stored procedure calls and created an extension method on DocumentClientException.

Reply:
Merged with that change.